client/authority-discovery/src/tests.rs

```diff
@@ -0,0 +1,367 @@
+use std::sync::{Arc, Mutex};
```

Comment:
This needs preamble. Why move the tests in the first place though? It's a common convention to just have an inline module.

Reply:
Besides less cognitive load, it'll be easier for me to format the code with `rustfmt` in a separate file :)